staging/src/k8s.io/apimachinery/pkg/util/managedfields/internal/managedfieldsupdater.go

```diff
@@ -72,11 +72,23 @@ func (f *managedFieldsUpdater) Apply(liveObj, appliedObj runtime.Object, managed
 	if err != nil {
 		return object, managed, err
 	}
+
 	if object != nil {
+		// non-nil object means the apply operation modified the object, so update the manager's timestamp
 		managed.Times()[fieldManager] = &metav1.Time{Time: time.Now().UTC()}
 	} else {
+		// nil object means the apply operation did not modify the input object
+		// clone the input object and return it without managed fields
 		object = liveObj.DeepCopyObject()
 		RemoveObjectManagedFields(object)
+
+		if _, managerInFields := managed.Fields()[fieldManager]; managerInFields {
```

Comment:
@jpbetz how does partitioning of field manager among subresources work? does managed.Fields() only return entries relevant to the subresource for this request?

Reply:
The field manager relies on the `GetResetFields()` ([example](https://github.com/kubernetes/kubernetes/blob/810e9e212ec5372d16b655f57b9231d8654a2179/pkg/registry/core/pod/strategy.go#L202)) of the subresource to indicate which fields the subresource applies to. 

'GetResetFieldsFilter()' was also added a in 1.32 to make it easier to declare fields for subresources ([example](https://github.com/kubernetes/kubernetes/blob/810e9e212ec5372d16b655f57b9231d8654a2179/pkg/registry/core/pod/strategy.go#L339C28-L343))